### ISSUE TITLE
Support the reference symbol feature of  i18n

### ISSUE THREAD

**Issue by tomoasleep** [glebm/i18n-tasks#150]
The default backend (Backend::Simple) of i18n gem has a feature that allows you to specify translations as a reference to another key.

For example, the following YAML translation data is given:

```yaml
en:
  sample: :foo.bar
  foo:
    bar:
      run:
        success: Success!
```

When you evaluate `I18n.translate('.sample.run.success')` it returns `'Success!'`.
This is because that [the backend interprets](https://github.com/svenfuchs/i18n/blob/v0.7.0/lib/i18n/backend/simple.rb#L78) the symbol value `:foo.bar` corresponding to the key `'.sample.run.success'` as
the reference of another key `'.foo.bar'`.

Currently i18n-tasks gem does not support the feature and reports the use of such symbol references as missing keys.

If i18n-task gem will support this feature, I will send a PR :-)
What do you think?

Here's a reproducible example project:

* https://github.com/tomoasleep/i18n-tasks-sample-app

**Comment by glebm** [comment]
Hi, @tomoasleep! Apologies for the late response. The goal of this project is to support as much of the i18n gem as possible, so I will merge a PR for this. :+1:

**Comment by tomoasleep** [comment]
@glebm Thanks for replying!
 I'm busy now and I'm sorry that It would take time..., but I will send a PR :+1:

**Comment by rstacruz** [comment]
any kind soul want to offer doing this?

**Comment by glebm** [comment]
Basic requirements / design:

1. Consider reference key usages equivalent to using the resolved version (similar to how we handle relative keys).
2. Detect reference keys as unused if the target as unused.
3. Detect reference keys as missing if the target is missing (a new special type of a missing key).
4. Copy reference keys as is on add-missing and translate-missing.

**Comment by glebm** [comment]
A very basic design that doesn't give i18n-tasks full awareness of reference keys, meaning only basic reporting, but achieving all of the basic requirements above:

Consider:

```yaml
en:
  reference: :resolved
  resolved:
    x: 1
```

1. Include all resolved references into `used_key_names`. What this means for the used references is:
  1. If a `reference` key is missing, it is reported as such.
  2. If a `resolved.*` key is missing, it is reported as such.
  3. Both `reference` and `resolved` are considered used.

  All unused references are also detected as unused.

  This achieves requirements (1) and (2).

2. Hard-code reference check into add-missing and translate-missing to always copy as-is.

**Comment by glebm** [comment]
In theory, references should be somewhat supported for `find`, `missing`, and `unused` at this point.

No support for `add-missing` and `translate-missing`, finding keys by their resolved name via `find`, and nice reporting yet.

If you need this feature, please try the version from the `reference-keys` branch and report here.

**Comment by glebm** [comment]
Keys can now be find by their resolved name via `i18n-tasks find`. Reporting also improved slightly.

**Comment by glebm** [comment]
Released in v0.9.3.
